crates/matrix-sdk-indexeddb/src/state_store/mod.rs

```diff
@@ -740,38 +754,55 @@ impl_state_store!({
 
         if !changes.redactions.is_empty() {
             let state = tx.object_store(keys::ROOM_STATE)?;
-            let room_info = tx.object_store(keys::ROOM_INFOS)?;
+            let room_infos = tx.object_store(keys::ROOM_INFOS)?;
 
             for (room_id, redactions) in &changes.redactions {
+                // TODO: Load room version lazily, only if it is necessary
+                let room_version = room_infos
+                    .get(&self.encode_key(keys::ROOM_INFOS, room_id))?
+                    .await?
+                    .and_then(|f| self.deserialize_event::<RoomInfo>(&f).ok())
+                    .and_then(|info| info.room_version().cloned())
+                    .unwrap_or_else(|| {
+                        warn!(?room_id, "Unable to find the room version, assume version 9");
+                        RoomVersionId::V9
+                    });
+
+                // Clean up raw events we saved for roominfo
+                if changes.room_infos.get(room_id).is_some() {
+                    // This roominfo has changed, so let's update the raw events
```

Comment:
I don't understand why we need this :/ It seems that a redaction would cause a change to the `RoomInfo`, so we'd convert it into a `RawRoomInfo` before persisting it, and we wouldn't need any special handling of redacted events?

Reply:
Sorry, I think I understand it now. So as I've suggested in a private channel, the duplication of the code applying the redaction to events both in `BaseRoomInfo` and `RawRoomInfo` is a bit problematic, and a hindrance in the long term. I wonder if we could regenerate a `BaseRoomInfo` from the `RawRoomInfo` every time we received an update to the `RawRoomInfo`, instead? That would keep the redaction code in a single place and avoid duplication of concerns.